Source/WebCore/platform/image-decoders/ScalableImageDecoder.cpp

```diff
@@ -68,72 +75,85 @@ static unsigned copyFromSharedBuffer(char* buffer, unsigned bufferLength, const
     return bytesExtracted;
 }
 
-bool matchesGIFSignature(char* contents)
+#if !PLATFORM(COCOA)
+static bool matchesGIFSignature(char* contents)
 {
     return !memcmp(contents, "GIF87a", 6) || !memcmp(contents, "GIF89a", 6);
 }
 
-bool matchesPNGSignature(char* contents)
+static bool matchesPNGSignature(char* contents)
 {
     return !memcmp(contents, "\x89\x50\x4E\x47\x0D\x0A\x1A\x0A", 8);
 }
 
-bool matchesJPEGSignature(char* contents)
+static bool matchesJPEGSignature(char* contents)
 {
     return !memcmp(contents, "\xFF\xD8\xFF", 3);
 }
 
+static bool matchesBMPSignature(char* contents)
+{
+    return !memcmp(contents, "BM", 2);
+}
+
+static bool matchesICOSignature(char* contents)
+{
+    return !memcmp(contents, "\x00\x00\x01\x00", 4);
+}
+
+static bool matchesCURSignature(char* contents)
+{
+    return !memcmp(contents, "\x00\x00\x02\x00", 4);
+}
+#endif
+
 #if USE(AVIF)
-bool matchesAVIFSignature(char* contents)
+#if USE(CG)
+static bool matchesAVIFSignature(char*, FragmentedSharedBuffer& data)
+{
+    auto sharedBuffer = data.makeContiguous();
+    auto cfData = sharedBuffer->createCFData();
+    auto imageSource = adoptCF(CGImageSourceCreateWithData(cfData.get(), nullptr));
```

Comment:
I think this means we will create two ImageSources for every non AVIF image: one here and one in ImageDecoderCG. How does affect PLT for example?

Reply:
Good point. I'll run PLT.

Reply:
Oh, following your advice above, I moved this code behind `USE(AVIF)` which isn't enabled on Ventura, which is where we test PLT. So I think your advice above solves both of these concerns 👍